### PR TITLE
Optional

### PR DESCRIPTION
**Pull request by diorcety**

### CODE REVIEW

File: pythran/pythonic/types/NoneType.hpp [hunk 1]
```diff
@@ -340,6 +316,23 @@ namespace pythonic
   }
 
   template <class T>
+  bool from_python<types::none<T>>::is_convertible(PyObject *obj)
```
Comment (serge-sans-paille) [inline pythran/pythonic/types/NoneType.hpp]: why do you need this? If I remember well, you either have a None, and then it's a none_type, or an object and then it's a plain object?

File: pythran/spec.py [hunk 2]
```diff
@@ -53,16 +53,17 @@ class SpecParser:
...
     # token <> regexp binding
-    t_CRAP = r'[^,:\(\)\[\]]'
+    t_CRAP = r'[^,:\(\)\[\]\*]'
     t_COMMA = r','
     t_COLUMN = r':'
     t_LPAREN = r'\('
     t_RPAREN = r'\)'
     t_RARRAY = r'\]'
     t_LARRAY = r'\['
+    t_OPTIONAL = r'\*'
```
Comment (serge-sans-paille) [inline pythran/spec.py]: I'd rather use either a `?` or the keyword `option` or `optional`, why the `*`?

File: pythran/tests/test_none.py [hunk 3]
```diff
@@ -36,3 +37,15 @@ def none_input(a):
...
+    def test_use_optional(self):
+        code = """
+            def none_input(a):
+                if a is None:
+                  return 0.0
+                else:
+                  return a
+            """
+        self.run_test(code, 2.0, none_input=[Optional(float64)])
+        self.run_test(code, None, none_input=[Optional(float64)])
```
Comment (serge-sans-paille) [inline pythran/tests/test_none.py]: ?? the syntax should be `Optional[float64]` I'm surprised this works.

Comment (diorcety) [inline pythran/tests/test_none.py]: I copy past the previous test :D

**Comment by serge-sans-paille** [comment]
Obsoleted by #659

**Pull request closed without merging**